src/libraries/System.IO.FileSystem/tests/File/GetSetTimes.cs

```diff
@@ -70,47 +73,200 @@ public async Task CreationTimeSet_GetReturnsExpected_WhenNotInFuture()
+        private static void SetCreationTimeUsingHandle(string path, DateTime creationTime)
         {
-            if (IOInputs.SupportsGettingCreationTime && (!requiresRoundtripping || IOInputs.SupportsSettingCreationTime))
+            using var fileHandle = OpenFileHandle(path);
+            File.SetCreationTime(fileHandle, creationTime);
+        }
+
+        private static DateTime GetCreationTimeUsingHandle(string path)
+        {
+            using var fileHandle = OpenFileHandle(path);
+            return File.GetCreationTime(fileHandle);
+        }
+
+        private static void SetCreationTimeUsingHandleUtc(string path, DateTime creationTime)
+        {
+            using var fileHandle = OpenFileHandle(path);
+            File.SetCreationTimeUtc(fileHandle, creationTime);
+        }
+
+        private static DateTime GetCreationTimeUsingHandleUtc(string path)
+        {
+            using var fileHandle = OpenFileHandle(path);
+            return File.GetCreationTimeUtc(fileHandle);
+        }
+
+        private static void SetLastAccessTimeUsingHandle(string path, DateTime creationTime)
+        {
+            using var fileHandle = OpenFileHandle(path);
+            File.SetLastAccessTime(fileHandle, creationTime);
+        }
+
+        private static DateTime GetLastAccessTimeUsingHandle(string path)
+        {
+            using var fileHandle = OpenFileHandle(path);
+            return File.GetLastAccessTime(fileHandle);
+        }
+
+        private static void SetLastAccessTimeUsingHandleUtc(string path, DateTime creationTime)
+        {
+            using var fileHandle = OpenFileHandle(path);
+            File.SetLastAccessTimeUtc(fileHandle, creationTime);
+        }
+
+        private static DateTime GetLastAccessTimeUsingHandleUtc(string path)
+        {
+            using var fileHandle = OpenFileHandle(path);
+            return File.GetLastAccessTimeUtc(fileHandle);
+        }
+
+        private static void SetLastWriteTimeUsingHandle(string path, DateTime creationTime)
+        {
+            using var fileHandle = OpenFileHandle(path);
+            File.SetLastWriteTime(fileHandle, creationTime);
+        }
+
+        private static DateTime GetLastWriteTimeUsingHandle(string path)
+        {
+            using var fileHandle = OpenFileHandle(path);
+            return File.GetLastWriteTime(fileHandle);
+        }
+
+        private static void SetLastWriteTimeUsingHandleUtc(string path, DateTime creationTime)
+        {
+            using var fileHandle = OpenFileHandle(path);
+            File.SetLastWriteTimeUtc(fileHandle, creationTime);
+        }
+
+        private static DateTime GetLastWriteTimeUsingHandleUtc(string path)
+        {
+            using var fileHandle = OpenFileHandle(path);
+            return File.GetLastWriteTimeUtc(fileHandle);
+        }
+
+        public override IEnumerable<TimeFunction> TimeFunctions(bool requiresRoundtripping = false, bool testHandles = true)
```

Comment:
These tests follow an inheritance pattern where the base class writes a test against some abstract methods. And then derived classes fill in the methods with a specific API (like a class for `File` static methods, and another class for `FileInfo` instance methods).

Following that same pattern, I think the `SafeHandle` API should probably be in its own class like `SafeHandle_GetSetTimes` and not added to `File_GetSetTimes`.